src/gameObjects/player.js

```diff
@@ -31,6 +34,74 @@ export default class Player {
       friction: 0.1,
       render: { sprite: { xOffset: 0.5, yOffset: 0.5 } },
     });
-    this.sprite.setExistingBody(compoundBody).setPosition(x, y).setIgnoreGravity(true).setOrigin(0.4, 0.6);
+    this.sprite
+      .setExistingBody(compoundBody)
+      .setPosition(x, y)
+      .setIgnoreGravity(true)
+      .setOrigin(0.4, 0.6);
+    this.reticle = this.scene.add.sprite(x, y, "target").setOrigin(0.5, 0.5);
+    this.reticle.setDepth(10);
+    this.addEvents();
+  }
+
+  addEvents() {
+    this.scene.events.on("update", this.update, this);
+  }
+
+  update() {
+    let velocityX = 0;
```

Comment:
**suggestion**: Consider extracting boundary calculations into a separate method for better readability and potential reuse.